src/infill.cpp

```diff
@@ -551,40 +507,26 @@ void Infill::generateZigZagInfill(Polygons& result, const coord_t line_distance,
  * Edit: the term scansegment is wrong, since I call a boundary segment leaving from an even scanline to the left as belonging to an even scansegment, 
  *  while I also call a boundary segment leaving from an even scanline toward the right as belonging to an even scansegment.
  */
-void Infill::generateLinearBasedInfill(const int outline_offset, Polygons& result, const int line_distance, const PointMatrix& rotation_matrix, ZigzagConnectorProcessor& zigzag_connector_processor, const bool connected_zigzags, coord_t extra_shift)
+void Infill::generateLinearBasedInfill(Polygons& result, const int line_distance, const PointMatrix& rotation_matrix, ZigzagConnectorProcessor& zigzag_connector_processor, const bool connected_zigzags, coord_t extra_shift)
 {
     if (line_distance == 0)
     {
         return;
     }
-    if (outer_contour.size() == 0)
+    if (inner_contour.empty())
     {
         return;
     }
 
     coord_t shift = extra_shift + this->shift;
 
-    if (outline_offset != 0 && perimeter_gaps)
-    {
-        const Polygons gaps_outline =
-            outer_contour.offset(outline_offset + infill_line_width / 2 + perimeter_gaps_extra_offset);
-        perimeter_gaps->add(outer_contour.difference(gaps_outline));
-    }
-
-    Polygons outline = outer_contour.offset(outline_offset + infill_overlap);
-
+    Polygons outline = inner_contour; //Make a copy. We'll be rotating this outline to make intersections always horizontal, for better performance.
     if (outline.size() == 0)
```

Comment:
Indeed, that is now superfluous. It was necessary previously because it used to do that inset here. Removed with e15d5cfd8e6db2ab85c8a451dbda859d98db4d94